lib/server/treatments.js

```diff
@@ -143,12 +143,31 @@ function storage (env, ctx) {
 
 function prepareData(obj) {
 
+  // Convert all dates to UTC dates
+
+  let d = new Date();
+
+  try {
+    if (obj.created_at) d = new Date(obj.created_at);
+  } catch (error) {
+    console.error(error);
```

Comment:
Is catching and "consuming" a format error really what we want? In case of bad format the current datetime is going to be used. That means that original event timestamp is forgotten and suddenly the event looks like it has occured right now. Wouldn't it be better in this case interrupt the action and finish the API command with 401 HTTP status error code?

Reply:
I would also agree with your suggestion in discussion and make created_at a required field. In general I think that being more strict about incoming data is better, because then you protect all other client apps that uses this data subsequently. Being too benevolent can lead to unpredictable weird issues like we are facing now. 
So I suggest to reply with 401 every time when created_at is missing or it is malformed.

Reply:
No, entries with bogus data should cause an error response from the server. Sadly we have so many layers of abstraction between this and the HTTP response that this is non-trivial to implement, unless I move this code to the HTTP API layer. Maybe that's the right thing to do...

Reply:
Changed